src/frequenz/sdk/ml/_model_manager.py

```diff
@@ -0,0 +1,130 @@
+# License: MIT
+# Copyright © 2024 Frequenz Energy-as-a-Service GmbH
+
+"""Load, update, monitor and retrieve machine learning models."""
+
+import asyncio
+import logging
+import pickle
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Generic, TypeVar, cast
+
+from frequenz.channels.file_watcher import EventType, FileWatcher
+from typing_extensions import override
+
+from frequenz.sdk.actor import BackgroundService
+
+_logger = logging.getLogger(__name__)
+
+T = TypeVar("T")
+
+
+@dataclass
+class _Model(Generic[T]):
+    """Represent a machine learning model."""
+
+    data: T
+    path: Path
+
+
+class ModelManager(BackgroundService, Generic[T]):
+    """Load, update, monitor and retrieve machine learning models."""
+
+    def __init__(self, model_paths: dict[str, Path], *, name: str | None = None):
+        """Initialize the model manager with the specified model paths.
+
+        Args:
+            model_paths: A dictionary of model keys and their corresponding file paths.
+            name: The name of the model manager service.
+        """
+        super().__init__(name=name)
+        self._models: dict[str, _Model[T]] = {}
+        self.model_paths = model_paths
+        self.load_models()
+
+    def load_models(self) -> None:
+        """Load the models from the specified paths."""
+        for key, path in self.model_paths.items():
+            self._models[key] = _Model(data=self._load(path), path=path)
+
+    @staticmethod
+    def _load(path: Path) -> T:
+        """Load the model from the specified path.
+
+        Args:
+            path: The path to the model file.
+
+        Returns:
+            T: The loaded model data.
+
+        Raises:
+            FileNotFoundError: If the model file does not exist.
+        """
+        if not path.exists():
+            raise FileNotFoundError(f"The model path {path} does not exist.")
```

Comment:
I would just let `open()` raise the error:
```suggestion
```

If you want to produce a more descriptive/specific error, I suggest creating a custom exception instead and chaining it to the orignal exception so users can get the full context:

```py
class ModelError(Exception):
    # You might want to actually add more details here, like `model_key` and `model_path`
    pass
```    
```suggestion
        try:
            with path.open("rb") as file:
                return cast(T, pickle.load(file))
        except Exception as err:
            raise ModelError(f"Could not load model {model_key} from {model_path}: {e}.", model_key, model_path) from err
```

Reply:
Done!